src/Main.elm

```diff
@@ -1,10 +1,177 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
+import Basics.Extra exposing (flip)
+import Browser
+import Data.Audience exposing (Audience)
 import Html exposing (Html)
+import Html.Attributes exposing (style)
+import Html.Events as Events
+import FolderInfo exposing (FolderInfo(..))
+import Maybe.Extra
+import String.Extra
+import Tree.Zipper as Zipper exposing (Zipper)
+import Zipper exposing (kids, tree)
 
 
-main : Html msg
+{- NOTE:    So I am not the most proud of the following code.
+            The styling is pretty arbitrary.
+            The html structure is pretty "medium rare" if you will...
+            In the real world app I think there would be some "company guidelines" or even some code to handle UI - I suppose.
+            So I kinda just went with it and had some fun.
+            The UI is kinda terrible, but I am assuming it may not be the main point of this challenge? Maybe I am wrong.
+            It has been some time since I have gotten my hands this dirty with actual component structures and styling.
+            I feel like it is just a matter of reading couple of articles about what is idiomatic nowadays or something like that.
+            And also - seeing the rest of the project - I feel like one can really infer a lot from some existing code and then just go and replicate that everywhere.
+            Hopefully y'all agree.
+-}
+
+
+
+type Msg
+    = Navigate (Zipper FolderInfo)
+
+
+
+-- I went with the Zipper alone - simply because it is convenient.
+type alias Model =
+    Zipper.Zipper FolderInfo
+
+
+init : Model
+init =
+    tree
+
+
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.sandbox { init = init, update = update, view = view }
+
+
+view : Model -> Html Msg
+view zipper =
+    let
+        (Info { audiences, name, id }) =
+            Zipper.label zipper
+
+        folders =
+            kids zipper
+
+        {- NOTE: I would not really write it like this in real project. Just thought I would have some fun and perhaps humor someone with a pointfree eye ;) -}
+        ifNotRoot =
+            (flip Maybe.andThen <| id) << always << Just
+
+        {- NOTE: I decided against it. One button is enough. -}
+        -- rootButton = viewGoButton "Back to TOP" <| Zipper.root zipper
+        -- toTop = ifNotRoot rootButton
+        goUp =
+            ifNotRoot <| viewGoUp zipper
+
+        folderTitle =
+            Html.span
+                [ style "padding" "15px"
+                , style "margin" "15px"
+                , style "background-color" "pink"
+                ]
+                [ Html.text <| "Now viewing: " ++ name ]
+
+        levelName =
+            ifNotRoot folderTitle
+
+        directories =
+            viewFolders folders
+
+        items =
+            if List.isEmpty audiences then
+                [{- EMPTY FOLDER -}]
+
+            else
+                List.map viewAudience audiences
```

Comment:
Is there any particular reason to add extra `if` check here? I believe List.map over empty list has same or very similar performance as  `List.isEmpty []`, so to me it's just additional code noise.

Reply:
I must have missed that. I was moving things around a lot in the last hour of working on it. And this is probably the case of cleaning, but not double checking.